### PR TITLE
v1.8.0 Release

### PR DESCRIPTION
**Pull request by YashTatkondawar**

### CODE REVIEW

File: docs/91_version_history.rst [hunk 1]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - An issue with forwarding data between two Fledge instances in different time zones has been resolved.
+       - A new API entry point has been added to the FogLAMP REST API to allow the removal of plugin packages.
+       - The notification service has been updated to allow for the delivery of multiple notifications in parallel.
+       - Improvements have been made to the handling of asset codes within the buffer in order to improve the ingest performance of FogLAMP. This is transparent to all services outside of the storage service and has no impact on the public APIs.
+       - Extra information has been added to the notification trigger such that trigger time and the asset that triggered the notification is included.
+       - A new configuration item type of “northTask” has been introduced. It allows the user to enter the name of a northTask in the configuration of another category within FogLAMP. 
+       - Data on multiple assets may now be requested in a a single call to the asset growing API within FogLAMP.
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:46]: repeated twice *a*

File: docs/91_version_history.rst [hunk 2]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - An additional API has been added to the asset browser to allow time bucketed data to be returned for multiple data points of multiple assets in a single call.
+       - Support has been added for nested readings within the reading data.
+       - Messages about exceeding the configured latency of the south service may be repeated when the latency is above the configured value for a period of time. These have now been replaced with a single emessgae when the latency is exceeded and another when the condition is cleared.
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:49]: typo *message*

File: docs/91_version_history.rst [hunk 3]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - The feedback provided to the user when a configuration item is set to an invalid value has been improved.
+       - Configuration items can now be marked as mandatory, this improves the user experience when configuring plugins.
+       - A new configuration item type, code, has been added to improve the user experience when adding code snippets in configuration data.
+       - Improvements have been made to the caching of configuration data within the core of FogLAMP.
+       - The logging of package installation has been improved.
+       - Additions have been adde to the public API to allow multiple audit log sources to be extracted extracted in a single API call.
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:55]: typo *added* and extracted word repeated twice

File: docs/91_version_history.rst [hunk 4]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - Improvements have been made to the checking of configuration items such that minimum, maximum values and string lengths are now checked.
+       - The plugin developers documentation has been updated to include a description building C/C++ south plugins.
+
+
+    - Bug Fix:
+
+       - Improvements have been made to the generation of the support bundle.
+       - An issue in the reporting of the task names in the fledge status script has been resolved.
+       - The purge by size (number of readings) would remove all data if the number of rows to retain was less than 1000, this has now been resolved.
+       - On occasions plugins woudl disappear from he list of available plugins, this has now been resolved.
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:70]: typo *would*, *the*

File: docs/91_version_history.rst [hunk 5]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - The support bundle functionality has been updated to provide information on the Python processes.
+       - The REST API incorrectly allowed configuration categories with a blank name to be created. This has now been prevented.
+       - Validation of minimum and maximum configuration item values was not correctly performed in the REST API, this has now been resolved.
+       - Nested objects within readings could cause the storage engine to fail and those readings to not be stored. This has now been resolved.
+       - On occasion shutting down a service may fail if the filters for that service have not been activated, this has now been resolved.
+       - An issue that cause notifications for asset whose names contain special characters has been resolved.
+       - The asset tracker was not correctly adding entries to the asset tracker, this has now been resolved.
+       - An intermittent issue that prevented the notification service being enabled on the Buster release on Raspberry Pi has been resolved.
+       - An intermittent problem that woudl prevent the north sending process to fail has been resolved.
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:83]: typo *would*

File: docs/91_version_history.rst [hunk 6]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - The counts in the user interface for notifications have been corrected.
+       - Asset data graphs are not able to handle correctly the transition between one day and the next. This is now resolved.
+
+
+- **Plugins**
+
+    - New Features:
+
+       - The existing set of OMF north plugins have been rationalised and replaced by a single OMF north plugin that is able to support the connector rely, PI Web API, EDS and OCS.
+       - A new notification rule plugin has been added that will trigger periodically if data is observed for a given asset. The trigger interval is defined in seconds. The plugin is primarily designed to be used with conditional forwarding as a means to sample data, although other uses might include a watchdog implementation for particular data streams.
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:128]: Which notification rule plugin?

File: docs/91_version_history.rst [hunk 7]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - When a Modbus TCP connection is closed by the remote end we fail to read a value, we then reconnect and move on to read the next value. On device with short timeout values, smaller than the poll interval, we fail the same reading every time and never get a value for that reading. The behaviour has been modified to allow us to retry reading the original value after re-establishing the connection.
+       - The OMF north plugin has been updated to support the released version of the OSIsoft EDS product as a destination for data.
+       - New functionality has been added to the north data to PI plugin when using PI Web API that allows the location in the PI Server AF hierarchy to be defined. A default location can be set and an override based on the asset name or metadata within the reading. The data may also be placed in multiple locations within the AF hierarchy.
+       - A new notification delivery plugin has been added that allows a north task to be triggered to send data for a period of time either side of the notification trigger event. This allows conditional forwarding of large amounts of data when a trigger event occurs.
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:132]: `notify-north` - Not applicable for fledge release

File: docs/91_version_history.rst [hunk 8]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - The asset notification delivery plugin has been updated to allow creation of new assets both for notifications that are triggered and/or cleared.
+       - The rate filter now allows the termination of sending full rate data either by use of an expression or by specifying a time in milliseconds.
+       - A new downsampling filter has been written that allows the rate of data to be reduced using one of a number of different algorithms.
+       - A new plugin that support the Digiducer 333D01 vibration sensor has been added.
+       - A new simple Python filter has been added that calculates an exponential moving average,
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:137]: `filter-downsample`, `south-digiducer`, `filter-simple-python` plugins are not applicable for fledge release

File: docs/91_version_history.rst [hunk 9]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - Some typos in the OPCUA south and north plugin configuration have been fixed.
+       - The OPCUA north plugin has been updated to support nested reading objects correctly and also to allow a name to be set for the OPCUA server. Thee have also been some stability fixes in the underlying OPCUA layer used by this and the south OPCUA plugin.
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:139]: typo *Thee* - These

File: docs/91_version_history.rst [hunk 10]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - The modbus map configuration now supports byte swapping and word swapping by use of the {{swap}} property of the map. This may take the values {{bytes}}, {{words}} or {{both}}.
+       - The people detection machine learning plugin now supports RTSP streams as input.
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:141]: `south-person-detection` is not applicable for fledge release

File: docs/91_version_history.rst [hunk 11]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - Support has been added for simple Python code to be entered into a filter that does not require all of the support code. This is designed to allow a user to very quickly develop filters with limited programming.
+       - Support has been added for filters written entirely in Python, these are full featured filters as supported by the C++ filtering mechanism and include dynamic reconfiguration.
+       - The foglamp-filter-expression filter has been modified to better deal with streams which contain multiple assets. It is now possible to use the syntax <assetName>.<datapointName> in an expression in addition to the previous <datapointName>. The result is that if two assets in the data stream have the same data point names it is now possible to differentiate between them.
+       - A new plugin to collect variables from Beckhoff PLC's has been written. The plugin uses the TwinCAT 2 or TwinCAT 3 protocols to collect specified variable from the running PLC.
+
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:149]: `south-beckhoff` is not applicable for fledge release

File: docs/91_version_history.rst [hunk 12]
```diff
@@ -25,6 +25,152 @@ Version History
...
+
+    - Bug Fix:
+
+       - An issue in the sending of data to the PI server with large values has been resolved.
+       - The playback south plugin was not correctly replaying timestamps within the file, this has now been resolved.
+       - Use of the asset filter in a north task could result in the north task terminating. This has now resolved.
+       - A small memory leak in the south service statistics handling code was impacting the performance of the south service, this is now resolved.
+       - An issue has been discovered in the Fair camera plugin with the validity attribute of the spot temperatures, this has now been resolved.
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:157]: typo - Flir

File: docs/91_version_history.rst [hunk 13]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - It was not possible to send data for the same asset from two different FogLAMP’s into the PI Server using PI Web API, this has now been resolved.
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:158]: Fledge's

File: docs/91_version_history.rst [hunk 14]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - The filter FogLAMP RMS Trigger was not able to be dynamically reconfigured, this has now been resolved.
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:159]: Fledge

File: docs/91_version_history.rst [hunk 15]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - If a filter in the north sending process increased the number of readings it was possible that the limit of the number of readings sent in a single block . The sending process will now ensure this can not happen.
+       - RMS filter plugin was not able to be dynamically reconfigured, this has now been resolved.
+       - The HTTP South plugin that is used to receive data from another FogLAMP instance may fail with some combinations of filters applied to the service. This issue has now been resolved.
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:162]: Fledge

File: docs/91_version_history.rst [hunk 16]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - The rule filter may give errors if expressions have variables not satisfied in the reading data. Under some circumstances it has been seen that the filter fails to process data after giving this error. This has been resolved by changes to make the rate filter more robust.
+       - Blank values for asset names in the south service may cause the service to become unresponsive. Blank asset names have now been correctly detected, asset names are required configuration values.
+       - A new version of the driver software for the USB-4704 Data Acquisition Module has been released, the plugin has been updated to use this driver version.
+       - The OPC UA North plugin might report incorrect counts for sent readings on some platforms, this has now been resolved.
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:166]: OPCUA

File: docs/91_version_history.rst [hunk 17]
```diff
@@ -25,6 +25,152 @@ Version History
...
+       - The simple Python filter plugin was not adding correct asset tracking data, this has now been updated.
```
Comment (ashish-jabble) [inline docs/91_version_history.rst:167]: N/A for fledge release

Comment (MarkRiddoch) [reply docs/91_version_history.rst:167]: Unfortunately we have no way to remove this, so will have to live with it for 1.8.

Comment (ashish-jabble) [reply docs/91_version_history.rst:167]: okay

**Pull request merged**